### PR TITLE
Amazon DocumentDB access part 2

### PR DESCRIPTION
**Pull request by greedy52**
Part 2
- ✅ (Part2) Discovery
- ✅ (Part2) URL checker
- ✅ (Part2) IAM configurator

See Part 1 for more details
- #43867

### CODE REVIEW

File: lib/cloud/aws/aws.go [hunk 1]
```diff
@@ -145,6 +145,21 @@ func AuroraMySQLVersion(cluster *rds.DBCluster) string {
...
+// https://docs.aws.amazon.com/documentdb/latest/developerguide/iam-identity-auth.html
+func IsDocumentDBClusterSupported(cluster *rds.DBCluster) bool {
+	ver, err := semver.NewVersion(aws.StringValue(cluster.EngineVersion))
+	if err != nil {
+		log.Errorf("Failed to parse DocumentDB engine version: %s", aws.StringValue(cluster.EngineVersion))
+		return false
+	}
+
+	minIAMSupportedVer := semver.New("5.0.0")
+	return !ver.LessThan(*minIAMSupportedVer)
```
Comment (nklaassen) [inline lib/cloud/aws/aws.go:160]: no need to parse a string
```suggestion
	return !ver.LessThan(semver.Version{Major: 5})
```

File: lib/cloud/aws/aws.go [hunk 2]
```diff
@@ -224,6 +239,13 @@ func IsRDSClusterAvailable(clusterStatus, clusterIndetifier *string) bool {
 	}
 }
 
+// IsDocumentDBClusterAvailable checks if the DocumentDB cluster is available.
+func IsDocumentDBClusterAvailable(clusterStatus, clusterIndetifier *string) bool {
+	// List of status values for DocumentDB is a subset of RDS's list:
+	// https://docs.aws.amazon.com/documentdb/latest/developerguide/monitoring_docdb-cluster_status.html
+	return IsRDSClusterAvailable(clusterStatus, clusterIndetifier)
```
Comment (nklaassen) [inline lib/cloud/aws/aws.go:246]:
```suggestion
func IsDocumentDBClusterAvailable(clusterStatus, clusterIdentifier *string) bool {
	// List of status values for DocumentDB is a subset of RDS's list:
	// https://docs.aws.amazon.com/documentdb/latest/developerguide/monitoring_docdb-cluster_status.html
	return IsRDSClusterAvailable(clusterStatus, clusterIdentifier)
```

File: lib/srv/db/cloud/resource_checker_url_aws.go [hunk 3]
```diff
@@ -272,3 +272,30 @@ func (c *urlChecker) checkOpenSearchEndpoint(ctx context.Context, database types
...
+	cluster, err := describeRDSCluster(ctx, rdsClient, meta.DocumentDB.ClusterID)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	databases, err := common.NewDatabasesFromDocumentDBCluster(cluster)
+	if err != nil {
+		c.log.Warnf("Could not convert DocumentDB cluster %q to database resources: %v.",
+			aws.StringValue(cluster.DBClusterIdentifier), err)
+
+		// services.NewDatabasesFromDocumentDBCluster maybe partially successful.
```
Comment (nklaassen) [inline lib/srv/db/cloud/resource_checker_url_aws.go:295]: Can you expand on why we want to return nil here?
```suggestion
		// common.NewDatabasesFromDocumentDBCluster may be partially successful.
```

**Review by nklaassen — APPROVED** [review]

File: lib/cloud/aws/aws.go (hunk 2 above)
Comment (smallinsky) [inline lib/cloud/aws/aws.go:246]: Can we just rename the IsRDSClusterAvailable to IsDBClusterAvailable ?

**Review by smallinsky — APPROVED** [review]

File: lib/srv/db/cloud/resource_checker_url_aws.go (hunk 3 above)
Comment (greedy52) [reply lib/srv/db/cloud/resource_checker_url_aws.go:295]: it shouldn't. will fix. thanks!

**Pull request merged**